### PR TITLE
Fix some documentation build warnings.

### LINKED ISSUE #1741: Can't make documentation without running make clean first
If you run `make` twice in the documentation directory without running `make clean` in-between, you see strange warning messages:
```
/Users/Adam/spack/lib/spack/docs/index.rst:44: WARNING: toctree contains reference to document u'package_list' that doesn't have a title: no link will be generated
/Users/Adam/spack/lib/spack/docs/index.rst:44: WARNING: toctree contains reference to document u'package_list' that doesn't have a title: no link will be generated
/Users/Adam/spack/lib/spack/docs/index.rst:44: WARNING: toctree contains reference to document u'package_list' that doesn't have a title: no link will be generated
/Users/Adam/spack/lib/spack/docs/index.rst:44: WARNING: toctree contains reference to document u'package_list' that doesn't have a title: no link will be generated
/Users/Adam/spack/lib/spack/docs/index.rst:44: WARNING: toctree contains reference to document u'package_list' that doesn't have a title: no link will be generated
generating indices... genindex/Users/Adam/spack/lib/spack/docs/index.rst:44: WARNING: toctree contains reference to document u'package_list' that doesn't have a title: no link will be generated
 py-modindex/Users/Adam/spack/lib/spack/docs/index.rst:44: WARNING: toctree contains reference to document u'package_list' that doesn't have a title: no link will be generated

writing additional pages... search/Users/Adam/spack/lib/spack/docs/index.rst:44: WARNING: toctree contains reference to document u'package_list' that doesn't have a title: no link will be generated
```
This warning message usually happens when you try to include a file without a title in the toctree, but package_list.rst does have a title, so I have no idea what's going on.

### PR DESCRIPTION
**Pull request by tgamblin**
Fixes #1741
@adamjstewart 

- Fix issue with package_list.py regeneration confusing Sphinx.
- Add -E option to avoid caching and make Sphinx happy.

### CODE REVIEW

File: lib/spack/docs/index.rst [hunk 1]
```diff
@@ -42,7 +42,6 @@ Table of Contents
 -----------------
 
 .. toctree::
-   :maxdepth: 2
```
Comment (adamjstewart) [inline lib/spack/docs/index.rst]: Why this change?

Comment (tgamblin) [inline lib/spack/docs/index.rst]: It was recommended as a fix for one of the indexing issues.  I believe it's also the default.  It didn't change the expansion depth when I looked at the output.

Comment (tgamblin) [inline lib/spack/docs/index.rst]: I don't mind taking it out because it didn't actually fix the indexing issue...
